tensorflow/contrib/opt/python/training/nadam_optimizer.py

```diff
@@ -80,7 +80,8 @@ def _apply_sparse_shared(self, grad, var, indices, scatter_add):
     with ops.control_dependencies([m_t]):
       m_t = scatter_add(m, indices, m_scaled_g_values)
       # m_bar = (1 - beta1) * g_t + beta1 * m_t
-      m_bar = m_scaled_g_values + beta1_t * m_t
+      state_ops.assign(m_t, m_t * beta1_t, use_locking=self._use_locking)
```

Comment:
This change does not make any sense to me. Assuming you wanted to maintain the original logic (m_bar = m_scaled_g_values + beta1_t * m_t), these two lines do not do that at all.

The first line creates an assign op that is never used, as the reference is not captured. So it will never be run.
The second line just does the same as line 81, adding m_scaled_g_values to m once more.

Have you tested this logic?